src/test/java/iq/base/AnnotationsTestsBase.java

```diff
@@ -165,32 +164,24 @@ public void initializeTests(File dir) {
                 Assert.fail("Program executed when not expected to");
             }
         }
-
         return context;
     }
 
-    protected void checkErrorCode(String errorOutput, String expectedErrorCode, String errorMetadata) {
-        if (errorOutput != null && expectedErrorCode != null)
-            Assert.assertTrue("Unexpected error code returned; Expected: " + expectedErrorCode +
-                    "; Error: " + errorOutput, errorOutput.contains(expectedErrorCode));
-        if (errorOutput != null && errorMetadata != null)
-            Assert.assertTrue("Unexpected metadata returned; Expected: " + errorMetadata +
-                    "; Error: " + errorOutput, errorOutput.contains(errorMetadata));
-    }
-
-    protected void checkExpectedOutput(String expectedOutput, RuntimeIterator runtimeIterator) {
-        Assert.assertTrue(true);
-    }
-
     private JsoniqParser.MainModuleContext parse(FileReader reader, JsoniqBaseVisitor visitor) throws IOException {
         JsoniqLexer lexer = new JsoniqLexer(new ANTLRInputStream(reader));
         JsoniqParser parser = new JsoniqParser(new CommonTokenStream(lexer));
         parser.setErrorHandler(new BailErrorStrategy());
 
         try {
-            JsoniqParser.ModuleContext unit = parser.module();
+            // the original
+            /*JsoniqParser.ModuleContext unit = parser.module();
             JsoniqParser.MainModuleContext main = unit.main;
-            visitor.visit(unit);
+            visitor.visit(unit);*/
+
```

Comment:
Sorry, got it. You now pass "main" to the visitor. That's the functional change.

Reply:
It would be nice if @wscsprint3r could take a look at this and clarify whether this was by design or a minor bug indeed. I can roll back to the original visiting pattern

Reply:
I will have a look :)

Reply:
I think the change is OK

Reply:
That's very good to hear. Could you explain why the behavior regarding the version differs when the MainModuleContext is visited instead of ModuleContext?